### PR TITLE
INT-760: Added Wavefront collector for K8s build support for Windows platform

### PR DESCRIPTION
**Pull request by gangadharaswamy**

### CODE REVIEW

**Comment by srinivas-kandula** [comment]
@gangadharaswamy , Could you explore this tool https://github.com/estesp/manifest-tool/releases to support multi arch releases.

File: Makefile [hunk 1]
```diff
@@ -62,6 +66,16 @@ ifneq ($(OVERRIDE_IMAGE_NAME),)
 	sudo docker tag $(PREFIX)/$(DOCKER_IMAGE):$(VERSION) $(OVERRIDE_IMAGE_NAME)
 endif
 
+#This rule need to be run on Windows 19 server with golang, cygwin, mktemp and docker installed.
+container_win: build
```
Comment (vikramraman) [inline Makefile:70]: Go allows you to cross-compile (i.e build `.exe` on Mac etc). We should add a `build_windows` target in the Makefile that emits a `.exe` file as relevant and depend on that for this target. Along the lines of:

```
GOOS=windows GOARCH=amd64 go build -ldflags "$(LDFLAGS)" -o $(OUT_DIR)/$(ARCH)/$(BINARY_NAME) ./cmd/wavefront-collector/
```

Not familiar with Windows container images. Can the docker image be built on a Mac (using `docker build`) with this Dockerfile?

That'll simplify the build side of things.

File: Makefile [hunk 2]
```diff
@@ -62,6 +66,16 @@ ifneq ($(OVERRIDE_IMAGE_NAME),)
...
+	cp $(OUT_DIR)/$(ARCH)/$(BINARY_NAME) $(TEMP_DIR)
+	cp deploy/docker/Dockerfile-win $(TEMP_DIR)/Dockerfile
+	docker build --pull -t $(PREFIX)/$(DOCKER_IMAGE):$(VERSION) $(TEMP_DIR)
```
Comment (vikramraman) [inline Makefile:73]: We should add a `-windows` suffix for these docker images.

File: deploy/multi-platform/README.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
+## Wavefront Collector images for multi-platform
```
Comment (vikramraman) [inline deploy/multi-platform/README.md:1]: Can you clarify what this is used for? Not clear why we need this.

Comment (gangadharaswamy) [reply deploy/multi-platform/README.md:1]: The manifest tool is to create a common image name for multiple platforms.
The collector image can be pulled using the same name `wavefronthq/wavefront-kubernetes-collector` for both Linux & Windows platform.
Reference doc: https://wavefront.atlassian.net/l/c/uk0Q21up

File: Makefile (hunk 1 above)
Comment (gangadharaswamy) [reply Makefile:70]: Windows docker images have to be built on a Windows machine.

File: Makefile (hunk 2 above)
Comment (gangadharaswamy) [reply Makefile:73]: We are using manifest-tool to create a common name for all platform docker images

**Comment by vikramraman** [comment]
Thanks for looking into this. We've settled on a different approach for now (running Collector as a deployment on Linux worker nodes) to get info from Windows worker nodes.

Closing this ticket, given the change in approach.

**Pull request closed without merging**